src/frontend/src/heartbeat/handler/invalidate_table_cache.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use catalog::helper::TableGlobalKey;
+use catalog::remote::KvCacheInvalidatorRef;
+use common_meta::error::Result as MetaResult;
+use common_meta::heartbeat::handler::{
+    HandleControl, HeartbeatResponseHandler, HeartbeatResponseHandlerContext,
+};
+use common_meta::instruction::{Instruction, InstructionReply, SimpleReply, TableIdent};
+use common_telemetry::error;
+
+#[derive(Clone)]
+pub struct InvalidateTableCacheHandler {
+    backend_cache_invalidtor: KvCacheInvalidatorRef,
+}
+
+impl HeartbeatResponseHandler for InvalidateTableCacheHandler {
+    fn is_acceptable(&self, ctx: &HeartbeatResponseHandlerContext) -> bool {
+        matches!(
+            ctx.incoming_message.as_ref(),
+            Some((_, Instruction::InvalidateTableCache { .. }))
+        )
+    }
+
+    fn handle(&self, ctx: &mut HeartbeatResponseHandlerContext) -> MetaResult<HandleControl> {
```

Comment:
I see a "handler pattern" here:
```rust
let Some(expected_instruction) = ctx.incoming_message.take() else { unreachable!() };
common_runtime::spawn_bg(async move {
    let result = do_some_work_with(expected_instruction).await;
    if let Err(e) = mailbox.send(convert_to_reply(result)).await { 
        error!(e; "Failed to send reply in mailbox"); 
    }
}
```
It occurred three times (plus this one), and maybe more in the future if we add more handlers to interact with meta. Can you come up with some refactoring to reduce the boilerplate codes? It's a bit challenging, you can just resolve this comment and add a todo if it's hard.